### DISCUSSION TITLE
zigbee2mqtt + SLZB-06

### DISCUSSION THREAD

**Discussion by Snake40** [Koenkk/zigbee2mqtt#23617] [General]
Bonjour.
I would like to test the following configuration:
- Supported NAS: DS220+,
- zigbee2mqtt under Docker,
- SLZB-06 dongle connected via Ethernet to my local network.
Everything I've read online leads to the conclusion: "easy to do"... But the world is cruel...
Here is the configuration of the Docker container on the NAS:
{
   "CapAdd" : null,
   "CapDrop" : null,
   "cmd" : "/sbin/tini -- node index.js",
   "cpu_priority" : 0,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : true,
   "enable_service_portal" : null,
   "enabled" : true,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "NODE_ENV",
         "value" : "production"
      },
      {
         "key" : "TZ",
         "value" : "Europe/Paris"
      }
   ],
   "exporting" : false,
   "id" : "33295dcbdfda50a9ba92d639622bea41f0d323ec7e17aa174b4a5bee13fcbe88",
   "image" : "koenkk/zigbee2mqtt:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "labels" : {},
   "links" : [],
   "memory_limit" : 0,
   "name" : "zigbee2mqtt",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 8085,
         "host_port" : 8085,
         "type" : "tcp"
      }
   ],
   "privileged" : false,
   "service_portals" : [],
   "shortcut" : {
      "enable_shortcut" : false,
      "enable_status_page" : false,
      "enable_web_page" : false,
      "web_page_url" : ""
   },
   "use_host_network" : false,
   "version" : 2,
   "volume_bindings" : [
      {
         "host_volume_file" : "/docker/zigbee2mqtt/data",
         "is_directory" : true,
         "mount_point" : "/data",
         "type" : "rw"
      }
   ]
}
Here is the configuration file "configuration.yaml" stored in the container's "/data" directory:

homeassistant: false
permit_join: true
mqtt:
    base_topic: zigbee2mqtt
    serveur:mqtt://192.168.1.100:1883
    user: xxx
    password: xxx
availability:
    active:
        timeout: 10
    passive:
        timeout: 1500
serial:
# Location of SLZB-06
    port: tcp://192.168.1.43:6638
    baudrate: 115200
    adapter: zstack
# Disable green led?
    disable_led: false
# Set output power to max 20
advanced:
    transmit_power: 20
frontend:
    port: 8085
	
And here is the execution log:

2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [31merror[39m:  z2m: Error: Error: No such file or directory, cannot open /dev/ttyACM0
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [31merror[39m:  z2m: Exiting...
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [31merror[39m:  z2m: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [31merror[39m:  z2m: Failed to start zigbee
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [31merror[39m:  z2m: Error while starting zigbee-herdsman
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [32minfo[39m:  zh:zstack:znp: Opening SerialPort with {"path":"/dev/ttyACM0","baudRate":115200,"rtscts":false,"autoOpen":false}
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [31merror[39m:  zh:zstack:znp: Failed to determine if path is valid: 'Error: ENOENT: no such file or directory, lstat '/dev/ttyACM0''
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [32minfo[39m:  z2m: Starting zigbee-herdsman (0.55.3)
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [32minfo[39m:  z2m: Starting Zigbee2MQTT version 1.39.1 (commit #e132316)
2024/08/14 10:41:05	stdout	[2024-08-14 10:41:05] [32minfo[39m:  z2m: Logging to console, file (filename: log.log)
2024/08/14 10:41:02	stdout	Starting Zigbee2MQTT without watchdog.
2024/08/14 10:41:02	stdout	Using '/app/data' as data directory

I am (very) rubbish when it comes to Linux... my understanding stops at everything I've been able to understand from my readings on the net.
Obviously the container is waiting for a connection with a USB dongle and... I don't know why ...
Could some of you give me some instructions so that the connection between the SLZB-06 and the zigbee2mqtt container is established?
Thank you in advance and have a good day.

**Comment by HobboRobin** [comment]
HEllo Mate,

From what i see in the log the configuration file is not correct. You have in the Configno port setup for the Adapter.

My config looks like this (i use also a ethernet Zigbee Adapter)

![image](https://github.com/user-attachments/assets/e7430053-74bf-497e-9398-bcfff40b79b1)

**Comment by Snake40** [comment]
Thank you for taking the trouble to send me this first response.
The SLZB-06 dongle has a web configuration page which gives in particular the details of the settings to be included in the "configuration.yaml" file, this is what I included between the lines "Location of SLB-06" and "transmit_power:20".
To be precise, in my message appear three lines in very bold and large characters, these are lines of comments in the .yaml file but which have apparently been interpreted and transformed by the messaging service.
In fact, everything happens as if the "configuration.yaml" file was not taken into account... I am therefore wondering where this file should be in my simplified configuration. Where is it located in your tree?

**Reply by HobboRobin** [reply]
Thats a complete different but good question. 
For me it is easy because i run it as a docker in unraid with path config.
From the docker itself that comes from github there should also be just a folder with /config
It should be excisting in your case because otherwise you get a message that there is no config file found or it creates one

**Comment by gam-st** [comment]
Check and correct you config file here:
"host_volume_file" : "/docker/zigbee2mqtt/data",

**Reply by Snake40** [reply]
Thanks for your help...
I suspect that I am making a mistake in declaring the volume(s) but even though I reread the Dockers documentation as well as some installation tutorials, I don't see how to do it.

Currently my directory structure is as follows:
- volume1
  - docker
    - mosquitto
      - config
        - mosquitto.conf
      - data
        ...
      - log
        ...
    - homeassistant
      - configuration.yml
      - docker-compose.yml
      ....
    - zigbee2mqtt
      - data
        - log
        - configuration.yml

Perhaps with this "photograph" of my directory structure you see my volume declaration error?

**Comment by gam-st** [comment]
@Snake40 
"host_volume_file" : "/docker/zigbee2mqtt/data",
"is_directory" : true,
"mount_point" : "/data",

Here is you mounted /docker/zigbee2mqtt/data on your HOST to the  /data in your z2m CONTAINER.
So your config should be located in /docker/zigbee2mqtt/data.

Do you see configuration.yaml here?

If yes, and z2m does not pick it up, you need to adjust destination path in container, because it is actually not '/data' but '/app/data'. 

There is no path '/data', thus your z2m container creates data inside with defaults and uses it.

Let me know if that helped

**Reply by Snake40** [reply]
Thank you for your help and apologies in advance for my "primary" questions...
I work from a docker compose, are the 3 lines at the beginning of your message to be put in the docker compose or are these SSH commands or another nature?
For now, I changed the docker compose and used the following volume declaration:

volumes:
    - /volume1/docker/zigbee2mqtt/data:/app/data

The zigbee2mqtt "configuration.yaml" file is in the "data" directory...
On this basis, the container was installed then it began a startup and shutdown sequence with each time the following error messages (different from those I copied in my first post):

2024/08/16 20:07:11	stderr	    at start (/app/index.js:130:14)
2024/08/16 20:07:11	stderr	    at Object.reRead (/app/lib/util/settings.ts:749:5)
2024/08/16 20:07:11	stderr	    at getInternalSettings (/app/lib/util/settings.ts:466:21)
2024/08/16 20:07:11	stderr	    at read (/app/lib/util/settings.ts:380:27)
2024/08/16 20:07:11	stderr	    at interpretValue (/app/lib/util/settings.ts:369:25)
2024/08/16 20:07:11	stderr	    at Object.read (/app/lib/util/yaml.ts:7:37)
2024/08/16 20:07:11	stderr	    at Object.readFileSync (node:fs:464:35)
2024/08/16 20:07:11	stderr	    at Object.openSync (node:fs:596:3)
2024/08/16 20:07:11	stderr	Error: ENOENT: no such file or directory, open '/app/data/... 3.yaml'
2024/08/16 20:07:10	stdout	Starting Zigbee2MQTT without watchdog.
2024/08/16 20:07:10	stdout	Using '/app/data' as data directory

The line below confuses me a little because the path does not seem to be a shortened path (in the sense of an overly long display which would be simplified) but rather the message as it is sent...

Error: ENOENT: no such file or directory, open '/app/data/... 3.yaml'

I searched the net but couldn't find anything constructive. Once again, I have the impression that the installation cannot find the configuration file.
